types/amazon-cognito-auth-js/index.d.ts

```diff
@@ -573,5 +590,65 @@ export class StorageHelper {
      * This is used to return the storage
      * @returns the storage
      */
-    getStorage(): Storage;
+    getStorage(): CognitoAuthStorage;
+}
+
+export interface CognitoAuthStorage extends Pick<Storage, 'setItem' | 'getItem' | 'removeItem' | 'clear'> {}
+
+export interface CookieStorageOptions {
+    /**
+     * Cookies domain (mandatory).
+     */
+    domain: string;
+
+    /**
+     * Cookies path (default: '/')
+     */
+    path?: string;
+
+    /**
+     * Cookie expiration (in days, default: 365)
+     */
+    expires?: number;
+
+    /**
+     * Cookie secure flag (default: true)
+     */
+    secure?: boolean;
+}
+
+export class CookieStorage implements CognitoAuthStorage {
+    /**
+     * Constructs a new CookieStorage object
+     * @param data Creation options.
+     */
+    constructor(data: CookieStorageOptions);
+
+    /**
+     * This is used to set a specific item in storage
+     * @param key the key for the item
+     * @param value the value
+     * @returns value that was set
+     */
+    setItem(key: string, value: unknown): string;
+
+    /**
+     * This is used to get a specific key from storage
+     * @param key the key for the item
+     * @returns the data item
+     */
+    getItem(key: string): string | null;
```

Comment:
```suggestion
    getItem(key: string): string | undefined;
```

https://github.com/amazon-archives/amazon-cognito-auth-js/blob/master/src/CookieStorage.js#L58
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/js-cookie/index.d.ts#L65